tests/json.rs

```diff
@@ -2,6 +2,23 @@ use slack_blocks::block_elements::BlockElement;
 use slack_blocks::blocks::Block;
 use slack_blocks::compose;
 
+fn check_no_nulls(json: &serde_json::Value) -> Result<(), ()> {
```

Comment:
There is **_nothing_** wrong with your implementation here. This is fine code that will work exactly as you predict.

Since you've mentioned you're relatively new to the rusty way of thinking, i figured this would be a good opportunity to share a mantra you'll hear fairly often:

Most of the time, `if/else` can be replaced with either a `match` statement or an `Option`, and that **can** result in code that's easier to reason about.

For example, here's another way you could implement this function:
```rust
fn check_no_nulls(json: &serde_json::Value) -> Result<(), ()> {
    use serde_json::Value;

    let check_many_no_nulls = |iter: Iterator<Item = Value>| -> Result<(), ()> {
        item
            .map(check_no_nulls)
            .collect::<Result<Vec<_>, _>>()
            .map(|_| {})
    };

    match json {
        Value::Object(map) => check_many_no_nulls(map.values()),
        Value::Array(arr) => check_many_no_nulls(arr.iter()),
        Value::Null => Err(()),
        _ => Ok(()),
    }
}
```

Let me know what you think of this! if the iterator stuff is confusing I added some comments in an expand section down here :point_down: 

<details>
<summary><b>Click to expand explanation</b></summary>

```rust
fn check_no_nulls(json: &serde_json::Value) -> Result<(), ()> {
    use serde_json::Value; // bring Value into scope

    // shorthand for checking an iterator of JSON values for nulls
    let check_many_no_nulls = |iter: Iterator<Item = Value>| -> Result<(), ()> {
        item
            .map(check_no_nulls) // For each item, check that it has no null props
            // This is a fancy trick that turns an Iterator of Results
            // into a big Result of a collection.
            // Note that this will `collect` into an Err on the
            // *first* Err it finds as it iterates.
            // If you want _all_ the Errs, you have to do some trickier stuff
            .collect::<Result<Vec<_>, _>>()
            // The above call gives us a Result of a Vec,
            // but we want to return unit (), and we don't
            // care about the units () in the Vec,
            // so I just use a map call with an empty body
            // to quickly get a Result over ().
            .map(|_| {})
    };

    match json {
        Value::Object(map) => check_many_no_nulls(map.values()),
        Value::Array(arr) => check_many_no_nulls(arr.iter()),
        Value::Null => Err(()),
        _ => Ok(()),
    }
}
```
</details>

Reply:
One revision I would make to this code if it was part of actual production code is replace the `collect` into `Vec` then empty map with a reduce or fold, that way we're not spinning up a collection of nothings just to throw it away